src/engine/Server.cpp

```diff
@@ -237,18 +250,36 @@ ad_utility::url_parser::ParsedRequest Server::parseHttpRequest(
       parsedRequest.parameters_ =
           ad_utility::url_parser::paramsToMap(query->params());
 
+      if (parsedRequest.parameters_.contains("query") &&
+          parsedRequest.parameters_.contains("update")) {
+        throw std::runtime_error(
+            R"(Request must only contain one of "query" and "update".)");
+      }
       extractQueryFromParameters();
+      if (parsedRequest.parameters_.contains("update")) {
+        if (parsedRequest.parameters_["update"].size() > 1) {
+          throw std::runtime_error(
+              "The parameter \"update\" must be set exactly once.");
+        }
+        parsedRequest.operation_ =
+            Update{parsedRequest.parameters_["update"].front()};
+        parsedRequest.parameters_.erase("update");
+      }
```

Comment:
This action of "we want to extract a parameter and expect there to only be one value for it" happens a handful of times (not only in `Server.cpp`). So maybe make a struct out of the type alias so that we can have this as a member function?

Reply:
If you want to , you can of course also do that. I just thought that here the whole thing (optional lookup, checkin only once, erasing) is completey duplicated, so a function for that would already help.
But using a free function std::optional<string> getAndEraseParameterCheckAtMostOnce(...) (name not super important) fixes the cverage and this function.